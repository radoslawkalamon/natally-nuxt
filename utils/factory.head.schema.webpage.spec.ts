import { factoryHeadSchemaWebpage, FactoryHeadSchemaWebpageType } from '@/utils/factory.head.schema.webpage'

describe('Utils / Factory / Head / Schema / Webpage', () => {
  test('shall match snapshot | { path, title, webpageType = WebPage }', () => {
    const args = {
      path: '/test/path/',
      title: 'Test title',
      webpageType: FactoryHeadSchemaWebpageType.WebPage
    }
    expect(factoryHeadSchemaWebpage(args)).toMatchSnapshot()
  })

  test('shall match snapshot | { category, path, title, webpageType = CollectionPage }', () => {
    const args = {
      category: 'Test category',
      path: '/path/test/',
      title: 'Title test',
      webpageType: FactoryHeadSchemaWebpageType.CollectionPage
    }
    expect(factoryHeadSchemaWebpage(args)).toMatchSnapshot()
  })

  test('shall match snapshot | { category, createdAt, path, title, webpageType = WebPage }', () => {
    const args = {
      category: 'Test category',
      createdAt: '2021-11-01T00:00:00+00:00',
      path: '/path/test/',
      title: 'Title test',
      webpageType: FactoryHeadSchemaWebpageType.WebPage
    }
    expect(factoryHeadSchemaWebpage(args)).toMatchSnapshot()
  })

  test('shall match snapshot | { category, path, title, updatedAt, webpageType = CollectionPage }', () => {
    const args = {
      category: 'Test category',
      path: '/path/test/',
      title: 'Title test',
      updatedAt: '2021-11-03T00:00:00+00:00',
      webpageType: FactoryHeadSchemaWebpageType.CollectionPage
    }
    expect(factoryHeadSchemaWebpage(args)).toMatchSnapshot()
  })

  test('shall match snapshot | { category, createdAt, path, title, updatedAt, webpageType = WebPage }', () => {
    const args = {
      createdAt: '2021-11-05T00:00:00+00:00',
      path: '/path/test/',
      title: 'Title test',
      updatedAt: '2021-11-06T00:00:00+00:00',
      webpageType: FactoryHeadSchemaWebpageType.WebPage
    }
    expect(factoryHeadSchemaWebpage(args)).toMatchSnapshot()
  })
})
