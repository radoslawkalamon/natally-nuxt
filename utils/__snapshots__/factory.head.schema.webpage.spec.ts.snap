// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utils / Factory / Head / Schema / Webpage shall match snapshot | { category, createdAt, path, title, updatedAt, webpageType = WebPage } 1`] = `
Object {
  "@id": "https://169cm.pl/path/test/#webpage",
  "@type": "WebPage",
  "about": Object {
    "@id": "https://169cm.pl/#organization",
  },
  "breadcrumb": Object {
    "@id": "https://169cm.pl/path/test/#breadcrumb",
  },
  "dateModified": "2021-11-06T00:00:00+00:00",
  "datePublished": "2021-11-05T00:00:00+00:00",
  "inLanguage": "pl-PL",
  "isPartOf": Object {
    "@id": "https://169cm.pl/#website",
  },
  "name": "Title test • 169cm.pl",
  "potentialAction": Array [
    Object {
      "@type": "ReadAction",
      "target": Array [
        "https://169cm.pl/path/test/",
      ],
    },
  ],
  "primaryImageOfPage": Object {
    "@id": "https://169cm.pl/path/test/#primaryimage",
  },
  "url": "https://169cm.pl/",
}
`;

exports[`Utils / Factory / Head / Schema / Webpage shall match snapshot | { category, createdAt, path, title, webpageType = WebPage } 1`] = `
Object {
  "@id": "https://169cm.pl/path/test/#webpage",
  "@type": "WebPage",
  "about": Object {
    "@id": "https://169cm.pl/#organization",
  },
  "breadcrumb": Object {
    "@id": "https://169cm.pl/path/test/#breadcrumb",
  },
  "datePublished": "2021-11-01T00:00:00+00:00",
  "inLanguage": "pl-PL",
  "isPartOf": Object {
    "@id": "https://169cm.pl/#website",
  },
  "name": "Title test • Test category • 169cm.pl",
  "potentialAction": Array [
    Object {
      "@type": "ReadAction",
      "target": Array [
        "https://169cm.pl/path/test/",
      ],
    },
  ],
  "primaryImageOfPage": Object {
    "@id": "https://169cm.pl/path/test/#primaryimage",
  },
  "url": "https://169cm.pl/",
}
`;

exports[`Utils / Factory / Head / Schema / Webpage shall match snapshot | { category, path, title, updatedAt, webpageType = CollectionPage } 1`] = `
Object {
  "@id": "https://169cm.pl/path/test/#webpage",
  "@type": "CollectionPage",
  "about": Object {
    "@id": "https://169cm.pl/#organization",
  },
  "breadcrumb": Object {
    "@id": "https://169cm.pl/path/test/#breadcrumb",
  },
  "dateModified": "2021-11-03T00:00:00+00:00",
  "inLanguage": "pl-PL",
  "isPartOf": Object {
    "@id": "https://169cm.pl/#website",
  },
  "name": "Title test • Test category • 169cm.pl",
  "potentialAction": Array [
    Object {
      "@type": "ReadAction",
      "target": Array [
        "https://169cm.pl/path/test/",
      ],
    },
  ],
  "primaryImageOfPage": Object {
    "@id": "https://169cm.pl/path/test/#primaryimage",
  },
  "url": "https://169cm.pl/",
}
`;

exports[`Utils / Factory / Head / Schema / Webpage shall match snapshot | { category, path, title, webpageType = CollectionPage } 1`] = `
Object {
  "@id": "https://169cm.pl/path/test/#webpage",
  "@type": "CollectionPage",
  "about": Object {
    "@id": "https://169cm.pl/#organization",
  },
  "breadcrumb": Object {
    "@id": "https://169cm.pl/path/test/#breadcrumb",
  },
  "inLanguage": "pl-PL",
  "isPartOf": Object {
    "@id": "https://169cm.pl/#website",
  },
  "name": "Title test • Test category • 169cm.pl",
  "potentialAction": Array [
    Object {
      "@type": "ReadAction",
      "target": Array [
        "https://169cm.pl/path/test/",
      ],
    },
  ],
  "primaryImageOfPage": Object {
    "@id": "https://169cm.pl/path/test/#primaryimage",
  },
  "url": "https://169cm.pl/",
}
`;

exports[`Utils / Factory / Head / Schema / Webpage shall match snapshot | { path, title, webpageType = WebPage } 1`] = `
Object {
  "@id": "https://169cm.pl/test/path/#webpage",
  "@type": "WebPage",
  "about": Object {
    "@id": "https://169cm.pl/#organization",
  },
  "breadcrumb": Object {
    "@id": "https://169cm.pl/test/path/#breadcrumb",
  },
  "inLanguage": "pl-PL",
  "isPartOf": Object {
    "@id": "https://169cm.pl/#website",
  },
  "name": "Test title • 169cm.pl",
  "potentialAction": Array [
    Object {
      "@type": "ReadAction",
      "target": Array [
        "https://169cm.pl/test/path/",
      ],
    },
  ],
  "primaryImageOfPage": Object {
    "@id": "https://169cm.pl/test/path/#primaryimage",
  },
  "url": "https://169cm.pl/",
}
`;
